### PR TITLE
feat: add support for any key in custom protocols via custom data loaders

### PR DESCRIPTION
**Pull request by hbredin**
BREAKING: make "uri" key mandatory in custom protocol configuration file

See the [introduction to custom data loaders](https://github.com/pyannote/pyannote-database-custom-loader/)

cc @PaulLerner @vbrignatz

### CODE REVIEW

File: pyannote/database/custom.py [hunk 1]
```diff
@@ -230,113 +148,234 @@ def resolve_path(path: Text, database_yml: Path) -> Path:
...
-    x = databases.pop('X', None)
-    if x is not None:
-        databases['X'] = x
 
-    # for each database
-    for database_name, tasks in databases.items():
+    """
+
+    if "uri" not in entries:
+        msg = f"Missing mandatory 'uri' entry in {database}.{task}.{protocol}.{subset}"
```
Comment (PaulLerner) [inline pyannote/database/custom.py:219]: Did you purposely renamed 'uris' to 'uri' here ?

Comment (hbredin) [reply pyannote/database/custom.py:219]: Yes: the point is to make sure keys in `database.yml` and keys in `ProtocolFile`s are consistent.

**Comment by hbredin** [comment]
I did not really pushed the "Template" tests.
Can you try with the latest commit -- this might have fixed it.

**Comment by PaulLerner** [comment]
Close! My last commit fixes an other issue (and my custom data loader now works fine)

**Pull request merged**